pkg/network/kubemacpool_test.go

```diff
@@ -0,0 +1,190 @@
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+		Context("When only RangeStart is configured", func() {
+			It("should return an error because both or none of the ranges must be configured", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: ""}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
+				Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+			})
+		})
+
+		Context("When only RangeEnd is configured", func() {
+			It("should return an error because both or none of the ranges must be configured", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "", RangeEnd: "02:00:00:FF:FF:FF"}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
+				Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+			})
+		})
+
+		Context("When RangeStart contains 7 octets instead of 6", func() {
+			It("should return an error because the mac address set for RangeStart is invalid", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "00:00:00:00:00:00:00", RangeEnd: "02:FF:FF:FF:FF:FF"}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
+				Expect(errorList[0].Error()).To(Equal("failed to parse rangeStart because the mac address is invalid"))
+
+			})
+		})
+
+		Context("When RangeEnd contains 7 octets instead of 6", func() {
+			It("should return an error because the mac address set for RangeEnd is invalid", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "FF:FF:FF:FF:FF:FF:FF"}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
+				Expect(errorList[0].Error()).To(Equal("failed to parse rangeEnd because the mac address is invalid"))
+			})
+		})
+
+		Context("When range end is lesser than its start", func() {
+			It("should return an error", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:ff:00:00", RangeEnd: "02:00:00:00:00:00"}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
```

Comment:
`"validation returned unexpected errors: %v", errorList`

What if the first one is the one we expected, but following are indicating what is wrong

Reply:
I'm not sure I get you here. do you mind clarifying what you meant?
Thanks.

Reply:
So you test that exactly **one** error has been returned, if not, you raise. The raised message however only includes the first error. But what if we got 0 errors? Then we panic. And if we get 2 or more errors, we don't show them to the user running the test suite.